packages/docregistry/src/context.ts

```diff
@@ -596,6 +596,13 @@ export class Context<
 
     try {
       await this._manager.ready;
+      if (!this.canSave) {
+        // File cannot be saved. The "save" command is disabled in the UI,
+        // but if the user tries to save anyway, act as though it succeeded.
+        this._saveState.emit('completed');
+        return Promise.resolve();
+      }
```

Comment:
It would be good to consider what effect it has on the logic for updating the model hash (which is needed to prevent spurious "File Changed" dialogs). This condition prevents updating `contentHash` as it is performed in `_maybeSave` method and this condition prevents ever reaching it. I _hope_ this would not be a problem for RTC because it now uses `fileChanged` signal to propagate new `contentHash` since:
- https://github.com/jupyterlab/jupyterlab/pull/16695
- https://github.com/jupyterlab/jupyter-collaboration/pull/337

I also _think_ that it would not make the situation worse for other drives either as in that case they would not update the hash as they could not save the content.

I think that all we need to do is document the side-effects when saving and when saving is skipped, especially if we are going to emit `saveState` even if the save did not happen.